gnd/src/main/java/com/google/android/gnd/model/mutation/ObservationMutation.kt

```diff
@@ -0,0 +1,88 @@
+package com.google.android.gnd.model.mutation
+
+import com.google.android.gnd.model.form.Form
+import com.google.android.gnd.model.observation.ResponseDelta
+import com.google.android.gnd.util.toImmutableList
+import com.google.common.collect.ImmutableList
+import java.util.*
+
+data class ObservationMutation(
+    override val id: Long? = null,
+    override val type: Type = Type.UNKNOWN,
+    override val syncStatus: SyncStatus = SyncStatus.UNKNOWN,
+    override val projectId: String = "",
+    override val featureId: String = "",
+    override val layerId: String = "",
+    override val userId: String = "",
+    override val clientTimestamp: Date = Date(),
+    override val retryCount: Long = 0,
+    override val lastError: String = "",
+    var observationId: String = "",
+    var form: Form? = null,
+    var responseDeltas: ImmutableList<ResponseDelta> = ImmutableList.of()
+) : Mutation() {
+
+    override fun toBuilder(): Builder {
+        return Builder().also {
+            it.observationId = this.observationId
+            it.form = this.form
+            it.responseDeltas = this.responseDeltas
+        }.fromMutation(this) as Builder
+    }
+
+    override fun toString(): String =
+        super.toString() + "deltas= $responseDeltas"
+
+    inner class Builder : Mutation.Builder<ObservationMutation>() {
+        var observationId: String = ""
+            @JvmSynthetic set
```

Comment:
Is there a plan to get rid of `@JvmSynthetic` once we're 100% Kotlin? If so, can we track the post-migrations TODOs in an issue?

Reply:
yeah we should be able to drop them post-migration